test/e2e/flowaggregator_test.go

```diff
@@ -142,6 +145,17 @@ func TestFlowAggregator(t *testing.T) {
 	defer teardownTest(t, data)
 	defer teardownFlowAggregator(t, data)
 
+	if testOptions.providerName == "kind" {
+		// Currently, in Kind clusters, OVS userspace datapath does not support
+		// packet statistics in the conntrack entries. Because of that Flow Exporter
+		// at Antrea agent cannot consider flows to be active and keep sending active
```

Comment:
I'm trying to understand the explanation. If the stats don't change, and the flow is not considered active, won't it trigger the idle flow timeout at the exporter?

Reply:
Yes, Flow Export triggers the idle flow timeout at the exporter and sends a flow record as part of point 1 in this comment. I will add more info there to capture the same. 
Flow exporter does nothing until the conntrack flow gets deleted from conntrack table. When the conntrack entry goes to `TIME_WAIT` state, then Flow Exporter seconds the second flow record.

When running multiple times. I am still seeing errors where we see only one flow record corresponding to `TIME_WAIT` state and the first record is not available. Trying to debug that.

Reply:
Investigated further. There are few issues that contribute to unexpected behavior. Explained them in the PR description and commit message. The key reason for the intermittent nature of the failure is go-ipfix collector filter logging using `since` option in `kubectl logs`. I checked complete logs in the failed cases and found the missing records. As we have source port as one of the gating parameters,  we do not require any filter in logging.

Reply:
I'm still a bit confused because `exporterInactiveFlowExportTimeout` is set to  1 second for these tests, so if the flow cannot be considered active, it feels like a new record would be sent every 1s

The `aggregatorInactiveFlowRecordTimeout` is set to 6 seconds, so isn't there a risk of receiving 3 records from the aggregator if we get "unlucky" given that we run iperf 12s?

I feel like the comment would be easier to understand if it showed the timestamps at which we expect records from the exporter and the aggregator to be sent (with the reason). Kind of like what was done for the comment above.

Reply:
>I'm still a bit confused because exporterInactiveFlowExportTimeout is set to 1 second for these tests, so if the flow cannot be considered active, it feels like a new record would be sent every 1s

At Flow Exporter we have `exporterInactiveFlowExportTimeout`, which is set to be 1 second as you mentioned. As the flow is considered to be idle, we expire the flow record after the idle time out. We wait for the flow to become active again before exporting new records (either stats should increase or the tcp state of the connection should change). It will only become active when the iperf traffic finishes and the connection is closed; Flow exporter sends another flow record at this time.

Will add the time stamps to the comment.

Reply:
Thanks. It's mostly my bad, I keep forgetting that we send a single idle flow record, then wait for the connection to die or resume.

Reply:
Thanks for the review. Hope the test becomes more stable now.